gammapy/datasets/core.py

```diff
@@ -200,12 +202,23 @@ def read(cls, filedata, filemodel):
         """
         from . import DATASETS
 
-        models = Models.read(make_path(filemodel))
-        data_list = read_yaml(make_path(filedata))
+        path = make_path(path)
+
+        if (path / filedata).exists():
+            filedata = path / filedata
+        else:
+            filedata = make_path(filedata)
+        if (path / filemodel).exists():
+            filemodel = path / filemodel
+        else:
+            filemodel = make_path(filemodel)
+
+        models = Models.read(filemodel)
+        data_list = read_yaml(filedata)
 
         datasets = []
         for data in data_list["datasets"]:
-            dataset = DATASETS.get_cls(data["type"]).from_dict(data, models)
+            dataset = DATASETS.get_cls(data["type"]).from_dict(data, models, path=path)
```

Comment:
To avoid code duplication in the various `from_dict` methods, isn't it simpler to format the full path inside the loop here. We can safely assume that `Datasets.read` should be the main way to read datasets from disk no?